### PR TITLE
AzureRT S55 PR #2 - Fix VM ListAll & Set

### PR DESCRIPTION
**Pull request by huangpf**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@huangpf__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (phuang). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/commands/arm/vm/vmClient._js [hunk 1]
```diff
@@ -509,10 +509,17 @@ __.extend(VMClient.prototype, {
     }
 
     // boot diagnostics storage URI
-    options.disableBootDiagnostics = options.disableBootDiagnostics ? true : false;
-    options.enableBootDiagnostics = options.enableBootDiagnostics ? true : (options.disableBootDiagnostics ? false : true);
-    options.enableBootDiagnostics = options.bootDiagnosticsStorageUri ? true : options.enableBootDiagnostics;
-    options.bootDiagnosticsStorageUri = options.disableBootDiagnostics ? '' : (options.bootDiagnosticsStorageUri ? options.bootDiagnosticsStorageUri : null);
+    if (options.disableBootDiagnostics || options.enableBootDiagnostics || options.bootDiagnosticsStorageUri) {
+      options.disableBootDiagnostics = options.disableBootDiagnostics ? true : false;
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vmClient._js]: use "!!(options.disableBootDiagnostics)"

**Comment by yugangw-msft** [comment]
this change looks good, could you do rebase and squash to get rid of unneeded commits such as those 7 "merge pull request". i will merge after that

**Comment by huangpf** [comment]
@yugangw-msft Please help take a look after the rebase.

**Pull request merged**